src/lmake.cc

```diff
@@ -166,15 +166,17 @@ namespace lmake {
             size_t single_pos = to_match.find("*");
             if(double_pos != std::string::npos) {
                 std::string result = lmake::func::find_recursive(to_match);
-
                 char* res = (char*) std::malloc((result.size() + 1) * sizeof(char));
                 std::strcpy(res, result.c_str());
                 res[result.size()] = '\0';
 
                 lua_pushstring(vm, res);
                 return 1;
             } else if(single_pos != std::string::npos) {
-                char* res = lmake::func::find(to_match);
+                std::string result = lmake::func::find(to_match);
+                char* res = (char*) std::malloc((result.size() + 1) * sizeof(char));
+                std::strcpy(res, result.c_str());
+                res[result.size()] = '\0';
```

Comment:
Could be changed into a smart pointer and remove Memory leak completelly. Need to look at it because shared pointer could be freed when noone holds a copy.

Reply:
Not really necessary to have a smart pointer as that memory will not be freed anyway until the process ends.